borsh-derive/src/internals/attributes/item/mod.rs

```diff
@@ -85,17 +83,39 @@ pub(crate) fn contains_use_discriminant(input: &ItemEnum) -> Result<bool, syn::E
     Ok(use_discriminant.unwrap_or(false))
 }
 
-pub(crate) fn contains_initialize_with(attrs: &[Attribute]) -> Option<Path> {
+pub(crate) fn contains_initialize_with(attrs: &[Attribute]) -> Result<Option<Path>, Error> {
     let mut res = None;
     let attr = attrs.iter().find(|attr| attr.path() == BORSH);
     if let Some(attr) = attr {
-        let _ = attr.parse_nested_meta(|meta| {
```

Comment:
parsing error was silenced here, allowing to declare following value for the attribute without any `syn::Error` and returning `None`

```rust
#[derive(BorshSerialize, BorshDeserialize, PartialEq, Debug)]
#[borsh(init={strange; blocky})]
struct A {
    lazy: Option<u64>,
}
```